pkg/didcomm/common/service/service.go

```diff
@@ -31,9 +41,10 @@ type DIDComm interface {
 
 // Header helper structure which keeps reusable fields
 type Header struct {
-	ID     string           `json:"@id"`
-	Thread decorator.Thread `json:"~thread"`
-	Type   string           `json:"@type"`
+	ID      string           `json:"@id"`
+	Thread  decorator.Thread `json:"~thread"`
+	Type    string           `json:"@type"`
+	Purpose string           `json:"~purpose"`
```

Comment:
This will need to be revisited.

Reply:
changed it to `[]string` as per RFC (https://github.com/hyperledger/aries-rfcs/pull/351) 
Are you sure it should be a map? Still create issue #1037 and added TODO